pkgs/applications/misc/octoprint/default.nix

```diff
@@ -136,14 +190,17 @@ let
                 frozendict
                 future
                 itsdangerous
+                immutabledict
                 jinja2
                 markdown
                 markupsafe
                 netaddr
                 netifaces
                 octoprint-filecheck
                 octoprint-firmwarecheck
+                octoprint-pisupport
                 pkginfo
+                pip
```

Comment:
Adding pip allows OctoPrint to fail somewhat gracefully, by recognizing it cannot write to it's out path.

```
2021-08-22 17:25:35,905 - octoprint.util.pip - INFO - Using "/nix/store/k1id2ixqxvq0gr6dqcaddjcx4dpnqxqs-python3.8-pip-21.1.3/bin/pip" as command to invoke pip
2021-08-22 17:25:35,905 - octoprint.util.pip - INFO - pip installs to /nix/store/7ssj1jid9dqikk513rvg340kpbxrv0p0-python3-3.8.11/lib/python3.8/site-packages (writable -> no), --user flag needed -> no, virtual env -> no
2021-08-22 17:25:35,905 - octoprint.util.pip - INFO - ==> pip ok -> NO!
2021-08-22 17:25:35,905 - octoprint.util.pip - ERROR - Cannot use this pip install, can't write to the install dir and also can't use --user for installing. Check your setup and the permissions on /nix/store/7ssj1jid9dqikk513rvg340kpbxrv0p0-python3-3.8.11/lib/python3.8/site-packages.
2021-08-22 17:25:35,906 - octoprint.plugin - ERROR - Error while calling plugin softwareupdate
Traceback (most recent call last):
  File "/nix/store/vbgil0cpv9bsixw5rldl069yx8wl2cji-python3.8-OctoPrint-1.6.1/lib/python3.8/site-packages/octoprint/plugin/__init__.py", line 271, in call_plugin
    result = getattr(plugin, method)(*args, **kwargs)
  File "/nix/store/vbgil0cpv9bsixw5rldl069yx8wl2cji-python3.8-OctoPrint-1.6.1/lib/python3.8/site-packages/octoprint/util/__init__.py", line 1941, in wrapper
    return f(*args, **kwargs)
  File "/nix/store/vbgil0cpv9bsixw5rldl069yx8wl2cji-python3.8-OctoPrint-1.6.1/lib/python3.8/site-packages/octoprint/plugins/softwareupdate/__init__.py", line 176, in on_after_startup
    self._check_environment()
  File "/nix/store/vbgil0cpv9bsixw5rldl069yx8wl2cji-python3.8-OctoPrint-1.6.1/lib/python3.8/site-packages/octoprint/plugins/softwareupdate/__init__.py", line 314, in _check_environment
    and get_comparable_version(versions["pip"])
  File "/nix/store/vbgil0cpv9bsixw5rldl069yx8wl2cji-python3.8-OctoPrint-1.6.1/lib/python3.8/site-packages/octoprint/util/version.py", line 170, in get_comparable_version
    version_string = normalize_version(version_string)
  File "/nix/store/vbgil0cpv9bsixw5rldl069yx8wl2cji-python3.8-OctoPrint-1.6.1/lib/python3.8/site-packages/octoprint/util/version.py", line 211, in normalize_version
    if "-" in version:
TypeError: argument of type 'NoneType' is not iterable
```

This means the install is still setup declaratively.